Client/game_sa/CVehicleSA.cpp

```diff
@@ -51,6 +52,40 @@ void _declspec(naked) HOOK_Vehicle_PreRender(void)
     }
 }
 
+bool CanProcessFlyingCarStuff(CHeliSAInterface* heliInterface)
+{
+    SClientEntity<CVehicleSA>* vehicle = pGame->GetPools()->GetVehicle((DWORD*)heliInterface);
+    if (!vehicle || !vehicle->pEntity)
+        return true;
+
+    return vehicle->pEntity->GetHeliRotorState();
+}
+
+void _declspec(naked) HOOK_CHeli_ProcessFlyingCarStuff()
```

Comment:
Can you make the hook not naked?

Reply:
This would result in poorer readability of the code, because then I would have to mix inline-assembly with C++ code. Naked hook is fine

Reply:
How so? Isnt a naked function just a normal function but without stack set up?

Reply:
It's best to avoid mixing inline assembly with C++ code whenever possible, and in this case, it is possible. A naked hook is the right solution here. If you look at the hooks in MTA, you'll see that almost every one of them is naked because it's the best approach for hooks. It's also safer than mixing assembly code with C++, which can cause issues.

For example, there was an issue with glass explosions where a switch instruction caused a crash because it used the same register as the game at the same time. With naked hooks, we can precisely control register and stack operations, minimizing the risk of such problems

Reply:
its not about mixing C++ with assembly but not using `naked` functions as they are deprecated in newer versions of C++. If we ever want to move on, we'd need to drop naked functions entirely. MTA code runs in C++14/17 version.
Naked functions are a big factor for this.

> For functions declared with the naked attribute, the compiler generates code without prolog and epilog code. You can use this feature to write your own prolog/epilog code sequences using inline assembler code [[1](https://learn.microsoft.com/en-us/cpp/cpp/naked-cpp?view=msvc-170)]

> [[2](https://github.com/multitheftauto/mtasa-blue/pull/3616#discussion_r1697460495)] We are trying to avoid inline assembly for simple function calls.
> - C++ is more readable and maintainable than assembly
> - It's much easier to write c++ definition than assembly for developers
> - It generates better code. Compiler cannot optimize functions with inline assembly
> - You can avoid hacks with static variables

Reply:
Naked functions may be less preferred in modern C++ and their use might be considered less idiomatic. However, in the case of hooks and low-level operations, naked functions can be useful because they allow for direct manipulation of assembly code, which is not always possible with pure C++.

Reply:
You can still access everything from a non-naked function. You just have stack set up, thats all.

Reply:
Indeed, in a function that is not declared as naked, you can still access various elements of the program, but it requires the compiler to generate additional code for stack setup (prolog and epilog). This can limit your control over precise register management and code manipulation.

In the case of hooks and low-level manipulation, having full control over the machine code is crucial. Naked functions allow for direct manipulation, which is harder to achieve in functions where the prolog and epilog are automatically generated by the compiler. Therefore, using naked functions is often more appropriate in such situations. In this particular case, it is unlikely that i will make the change to a non-naked function.

Reply:
> We are trying to avoid inline assembly for simple function calls.

I mean only situations, when you need only a function call from c++ code.

> not using naked functions as they are deprecated in newer versions of C++. 

They aren't. Where did you find it?
Naked functions aren't 'real' functions. Naked function is a pure assembly without prologue, epilogue and **_return_**.
Function calling conversations aren't applied to naked functions, so you have full control there

> MTA code runs in C++14/17 version. Naked functions are a big factor for this.

Only 'Game SA' and 'Multiplayer SA' projects. It's a bug (?) in MSVC

Reply:
> They aren't. Where did you find it? Naked functions aren't 'real' functions. Naked function is a pure assembly without prologue, epilogue and **_return_**. Function calling conversations aren't applied to naked functions, so you have full control there

Naked functions are not supported in latest versions of C++ afaik. It throws error when you try to compile naked function in C++17 or c++20

Reply:
from premake5.lua
```
	-- HACK(Jusonex): Temp fix for ebp not being set in naked functions
	-- VS2019 changed the evaluation order as required by the C++17 standard
	-- which broke all functions marked with _declspec(naked) that call C++ code
	-- Falling back to the old, C++14 implementing fixes this
```

Reply:
Yeah this